### ISSUE TITLE
JSHint

### ISSUE THREAD

**Issue by Silverwolf90** [ArgentumVir/MullToZero#2]
I highly suggest you start using JSHint. It's a tool for static code analysis for Javascript. It catches errors you make while coding in Javascript and enforces a stricter less error-prone style. I would never program Javascript without it, it's a critical part of my workflow.

Examples of things it catches:
* Unused variables
* Accidental globals
* Trying to use a variable which doesn't exist
* enforcing usage of `===` over `==`

All within your text editor without having to test the page.

To use JSHint from within Sublime Text you'll need the SublimeLinter package and the JShint plugin for SublimeLinter. You should be able to install both from Sublime Text's package control (which does not come built in to Sublime and you must install independently [which if you haven't done yet you absolutely should, Sublime has a great plugin ecosystem])

**Comment by Silverwolf90** [comment]
Linting example within Sublime. The highlighted bits are the places where there are errors.

![image](https://cloud.githubusercontent.com/assets/1631625/8510940/aa953c9a-22f8-11e5-9f48-babda11f5c7a.png)

And here are the errors: 
```
line 25, col 18, Expected '===' and instead saw '=='. (W116)
line 25, col 35, Expected '===' and instead saw '=='. (W116)
line 25, col 53, Expected '===' and instead saw '=='. (W116)
line 25, col 73, Expected '===' and instead saw '=='. (W116)
line 25, col 98, Expected '===' and instead saw '=='. (W116)
line 25, col 125, Expected '===' and instead saw '=='. (W116)
line 37, col 47, Expected '===' and instead saw '=='. (W116)
line 42, col 21, Expected '===' and instead saw '=='. (W116)
line 52, col 6, Missing semicolon. (W033)
line 14, col 39, '$route' is defined but never used. (W098)
line 41, col 93, 'config' is defined but never used. (W098)
line 41, col 84, 'headers' is defined but never used. (W098)
line 41, col 76, 'status' is defined but never used. (W098)
line 46, col 95, 'config' is defined but never used. (W098)
line 46, col 86, 'headers' is defined but never used. (W098)
line 46, col 78, 'status' is defined but never used. (W098)
line 46, col 72, 'data' is defined but never used. (W098)
```

**Comment by Silverwolf90** [comment]
I just added and pushed a `.jshintrc` file to the repository. It's a configuration file for JSHint with the settings that I generally use. After you install JSHint it will automatically try and find a config file in the root of whatever project your're running it on.

So make sure when you do a `git pull` to get the latest changes.

**Comment by ArgentumVir** [comment]
Sounds useful, do I usually want to use === instead of == just because it's a stricter comparison?

**Comment by Silverwolf90** [comment]
Yes. `===` is less prone to bugs that way. In general, the more constraints on a piece of code, the easier it is to reason about. If you accept messy input you have to worry about way more cases than if you don't accept messy input. If you need to sanitize your data (for instance, converting strings to numbers), then do so in a precious step. But don't let it happen implicitly or else you're much more likely to have weird bugs.

But it's important to understand the concept of "truthy" and "falsy" in javascript. There's usually no reason to do explicit boolean comparisons.

```javascript
if (isConditionTrue === true) {
    // do thing
}
```
instead you can just do this
```javascript
if (isConditionTrue) {
   // do thing
}
```
You can do the same thing if you want to test for the existence of an object
```javascript
var object = maybeGetObject();
if (object) {
   // do thing with object
}
```
